File: src/test/java/org/nyxlang/lexer/LexerStatementTest.java
package org.nyxlang.lexer;

import org.junit.jupiter.api.Test;
import org.nyxlang.TestSuit;
import org.nyxlang.lexer.token.NewLineToken;
import org.nyxlang.lexer.token.StatementSeparatorToken;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;


public class LexerStatementTest extends TestSuit {

    @Test
    void shouldSeparateWithSemicolon() {
        var tokens = lexer.tokenize("true;true;false");
        assertEquals(5, tokens.length);
        assertSame(StatementSeparatorToken.class, tokens[1].getClass());
        assertSame(StatementSeparatorToken.class, tokens[3].getClass());
    }

    @Test
    void shouldSeparateWithNewLine() {
        var tokens = lexer.tokenize("true\ntrue\nfalse");
        assertEquals(5, tokens.length);
        assertSame(NewLineToken.class, tokens[1].getClass());
        assertSame(NewLineToken.class, tokens[3].getClass());
    }
}

File: src/test/kotlin/org/nyxlang/lexer/LexerStatementTest.kt
package org.nyxlang.lexer

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test
import org.nyxlang.TestSuit
import org.nyxlang.lexer.token.TokenType

class LexerStatementTest : TestSuit() {

    @Test
    fun shouldSeparateWithSemicolon() {
        val tokens = tokenize("true;true;false")
        assertEquals(6, tokens.size)
        assertSame(TokenType.SEPARATOR, tokens[1].kind)
        assertSame(TokenType.SEPARATOR, tokens[3].kind)
    }

    @Test
    fun shouldSeparateWithNewLine() {
        val tokens = tokenize("true\ntrue\nfalse")
        assertEquals(6, tokens.size)
        assertSame(TokenType.NEW_LINE, tokens[1].kind)
        assertSame(TokenType.NEW_LINE, tokens[3].kind)
    }
}